src/Raven.Server/Web/Authentication/AdminCertificatesHandler.cs

```diff
@@ -401,6 +505,57 @@ public Task GetAll()
             }
         }
 
+        [RavenAction("/admin/certificates/export", "GET", AuthorizationStatus.ClusterAdmin)]
+        public Task GetClusterCertificates()
+        {
+            if (IsClusterAdmin() == false)
+                throw new InvalidOperationException("Cannot export certificate collection. This is a 'Cluster Admin' only operation.");
+            
+            var start = GetStart();
+            var pageSize = GetPageSize();
+
+            var collection = new X509Certificate2Collection();
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                List<Tuple<string, BlittableJsonReaderObject>> allItems = null;
+                try
+                {
+                    allItems = ServerStore.Cluster.ItemsStartingWith(context, Constants.Certificates.Prefix, start, pageSize).ToList();
+                    var clusterNodes = allItems.Select(item => JsonDeserializationServer.CertificateDefinition(item.Item2))
+                        .Where(certificateDef => certificateDef.SecurityClearance == SecurityClearance.ClusterNode)
+                        .ToList();
+
+                    if (clusterNodes.Count == 0)
+                        throw new InvalidOperationException("Cannot get ClusterNode certificates, there should be at least one but it doesn't exist. This shouldn't happen!");
+
+                    foreach (var cert in clusterNodes)
+                    {
+                        var x509Certificate2 = new X509Certificate2(Convert.FromBase64String(cert.Certificate));
+                        collection.Import(x509Certificate2.Export(X509ContentType.Cert));
+                    }
+                }
+                finally
+                {
+                    if (allItems != null)
+                        foreach (var cert in allItems)
+                            cert.Item2?.Dispose();
+                }
+            }
+
+            var pfx = collection.Export(X509ContentType.Pfx);
+
+            var contentDisposition = "attachment; filename=" + Uri.EscapeDataString("ClusterCertificatesCollection") + ".pfx";
```

Comment:
Why are you calling the Escape here? This is static data, no?

Reply:
yes